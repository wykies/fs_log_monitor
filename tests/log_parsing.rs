use std::path::{Path, PathBuf};

use chrono::Local;
use fs_log_monitor::{build_err_msg_from_logs, process_logs_folder, AppState};

fn samples_folder() -> PathBuf {
    Path::new("tests").join("sample_logs")
}

#[test]
fn no_files_pass_filter() {
    let before_app_state_created = Local::now().naive_local();
    let mut actual = AppState::new(samples_folder());
    let expected = actual.clone();
    assert!(
        actual.latest_log_datetime() >= before_app_state_created,
        "date assumed to be now or later so that the sample logs from the past should not be included"
    );

    let log_infos = process_logs_folder(&mut actual).unwrap();
    assert!(
        log_infos.is_empty(),
        "all samples should be in the past and filtered out"
    );

    assert_eq!(actual, expected, "no changes expected to AppState created")
}

#[test]
fn output_snapshot() {
    let mut app_state = AppState::new_with_min_dates(samples_folder());

    let logs_infos = process_logs_folder(&mut app_state).unwrap();
    insta::assert_ron_snapshot!(logs_infos);
    insta::assert_ron_snapshot!(app_state, {
        ".last_alive_msg" => "date_time",
    });

    let msg = build_err_msg_from_logs(logs_infos);
    insta::assert_snapshot!(msg);
}
