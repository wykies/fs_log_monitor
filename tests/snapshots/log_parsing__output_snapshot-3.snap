---
source: tests/log_parsing.rs
expression: msg
---
---
[Error] 2024-10-15 09:28:45 errors: 1
Cannot open file "/home/bob/missing_file.pdf".; ENOENT: No such file or directory [stat]
---
[Error] 2024-11-08 14:09:13 errors: 4
Cannot read file "ProfileName:/sample@gmail.com/bob/logs.7z".; CURLE_OPERATION_TIMEDOUT: Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds [curl_easy_perform]
Cannot delete file "/home/bob/logs-ad8f.ffs_tmp".; ENOENT: No such file or directory [unlink]
Cannot read file "ProfileName:/sample@gmail.com/bob/log2.7z".; CURLE_OPERATION_TIMEDOUT: Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds [curl_easy_perform]
Cannot delete file "/home/bob/log2-62c0.ffs_tmp".; ENOENT: No such file or directory [unlink]
---
[Stopped] 2024-11-08 14:50:21 errors: 1
Stopped
---
3 logs with 6 error and warnings
---
